changelog.d/15905.doc

```diff
@@ -0,0 +1 @@
+Clarified some of the behaviour of the `synapse` vs `root` logger contexts. Contributed by @semiviral.
```

Comment:
> Clarified some of the behaviour of the `synapse` vs `root` logger contexts

I feel like I didn't learn anything more from these changes? What's the difference between `synapse` vs `root`?

Why can't we just suggest using the `root` version as a sane default that works regardless of workers or not?

Reply:
> (the loggers: synapse: logging context seems to only exist on the primary Synapse instance, and not its workers).
>
> *-- https://github.com/matrix-org/synapse/pull/15905#issuecomment-1628514566*

Your comment on the PR seems to have some of these details but I feel like I don't learn any of this from the docs you're trying to clarify.

Reply:
I agree, it's honestly pretty difficult to gauge the full scope of what I should be changing, since the logging section of our documentation is so bare as-is. I think larger changes are needed to fully express the changes I want to make.

I'm not, however, sure if it's okay to just submit a PR that makes sweeping changes to the documentation at-will? It seems like that could be unwise to do.

Additionally, I've done a little more testing with the remote logger, and have summarily lost all confidence in my understanding of it, and the Synapse logging context in general. It seems entirely random what Synapse itself wants to log vs what the `root` context logs. I'll need to actually dive into the source code to understand the behaviour I'm witnessing—it definitely doesn't feel correct as is.